core/API/DataTablePostProcessor.php

```diff
@@ -146,6 +147,23 @@ private function convertSegmentValueToSegment(DataTableInterface $dataTable)
         return $dataTable;
     }
 
+    /**
+     * @param DataTableInterface $dataTable
+     * @return DataTableInterface
+     */
+    public function applyArchiveStateFilter(DataTableInterface $dataTable): DataTableInterface
+    {
+        $fetchArchiveState = Common::getRequestVar('fetch_archive_state', '0', 'string', $this->request);
+
+        if ('0' === $fetchArchiveState) {
```

Comment:
I guess this parameter is meant to be some sort of boolish (0 / 1). We should maybe directly use the newer Request class here:
```suggestion
        $fetchArchiveState = (new \Piwik\Request($this->request))->getBoolParameter('fetch_archive_state', false);

        if (false === $fetchArchiveState) {
```